### PR TITLE
Zone Refactor - Decouples Z level properties to reservations

### PR DESCRIPTION
**Pull request by Azarak**
## About The Pull Request
All properties based on z-levels were moved to new `sub_map_zone`'s, which represents the level now, they can be defined with any boundaries, and crosslinked, multi-z linked freely. Those sub_map_zones are wrapped in `map_zone`'s which represent a collection of them and hold information such as affecting gravity generators, day/night controllers, weather controllers, tcomm information etc., They can be tied to overmap objects too.

All things needing to check z_level traits now check the sub_map_zone traits (probably I didn't catch all)
All things checking boundaries of z_levels now check boundaries of map zones (probably I didn't catch all)
Tcomm messages now pass map_zones instead of z levels, meaning you don't need relays for every single multi-z level
Planetary atmospheres now pull the information from the map zone, instead of the z level
GPS's now show 4 coordinates, relative X, relative Y, map zone ID, sub map zone ID (unique only to map zone)
Probably a couple changes I forgot about, but basically all z_level based stuff -> sub map zones

The good:
-Being able to define the zone reservations freely allows us to create a couple smaller "z levels" inside one single z level. Which can be already utilized to save memory for Bearcat, and a lot of the exploration z levels could be smaller, but more of them (I intend to do the bearcat change in an upcoming PR, need to write a space manager for it too first)
-With being able to define smaller memory efficient zones, we can look into implementing a variety of less-impactful levels with much less cost
-Map zones wrapping sub zones allows for easy bundling of information for them and better manipulation. (tcomms, grav gen, day/night etc.)
-Pinpointers and crew monitoring consoles could easily read from the sub zones of the map zone, to work multiz
-Soon I'll look into dynamically creating and removing those zones, so we can have many cheap dynamic exploration levels.

The bad:
-There is an increased cost of looking up the level properties, as the code needs to check boundaries instead of directly translating the z value to the z-level (I'll write a lookup optimisation after I'm done writing the space manager, since it's related)

## Changelog
:cl:
refactor: Refactored how level-based properties are managed, allowing for much more flexible level reservations
/:cl:

### CODE REVIEW

**Comment by atakiya** [comment]
> Sorry, it's a habit because CI used to not run on Drafts long time ago (and maybe TGS wouldnt recognize drafts as open PR's too i think?)

No worries.

Neither was ever an issue since this project started. but I can understand habits of course ^^

File: code/modules/admin/verbs/mapping.dm [hunk 1]
```diff
@@ -314,63 +313,3 @@ GLOBAL_VAR_INIT(say_disabled, FALSE)
...
-
-	messages += "<table border='1'>"
-	for(var/y in max_y to min_y step -1)
-		var/list/part = list()
-		for(var/x in min_x to max_x)
-			part += "[grid[x - min_x + 1][y - min_y + 1]]"
-		messages += "<tr><td>[part.Join("</td><td>")]</td></tr>"
-	messages += "</table>"
-
-	to_chat(src, messages.Join(""), confidential = TRUE)
```
Comment (MissFox0810) [inline code/modules/admin/verbs/mapping.dm:376]: Are you adding a Debug verb to replace the zlevel debug with a map level debug? (If you have just point out the file)

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 2]
```diff
@@ -658,6 +658,7 @@
 	icon = 'icons/obj/structures.dmi'
 	icon_state = "ladder00"
 
+/* TODO: re-do later
```
Comment (MissFox0810) [inline code/modules/mining/lavaland/necropolis_chests.dm:661]: Have you created an issue to come back to this so its not forgotten?

**Review by MissFox0810 — COMMENTED** [review]
Just some comments around debug verb removal. and a todo

File: code/modules/admin/verbs/mapping.dm (hunk 1 above)
Comment (Azarak) [reply code/modules/admin/verbs/mapping.dm:376]: A lot of the things that could go wrong no longer can go wrong, and all the information one would want to check, can be easily looked up in SSmapping by looking at map zones.

File: code/modules/mining/lavaland/necropolis_chests.dm (hunk 2 above)
Comment (Azarak) [reply code/modules/mining/lavaland/necropolis_chests.dm:661]: Good catch, I forgotten about this. I think I maybe should just remove this structure as it isn't very uh.. good and it doesn't make much sense either. I think a magic ladder that would connect all ruin levels, all planetary levels and the station is bad for gameplay

File: code/modules/admin/verbs/mapping.dm (hunk 1 above)
Comment (Azarak) [reply code/modules/admin/verbs/mapping.dm:376]: If you want I can write a verb that'd just compile a status of map zones with informations regarding them

**Comment by Azarak** [comment]
Debug z levels is now retained and jacobs ladder has been removed.

**Comment by Azarak** [comment]
Also added a debug print for map zone infos
obraz

**Review by atakiya — APPROVED** [review]

**Pull request merged**